src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -203,4 +198,87 @@ private function resetPagination()
             ldap_set_option($con, \LDAP_OPT_SERVER_CONTROLS, $ctl);
         }
     }
+
+    /**
+     * Sets LDAP pagination controls.
+     *
+     * @param resource $con
+     *
+     * @return bool
+     */
+    private function controlPagedResult($con, int $pageSize, string $cookie)
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            return ldap_control_paged_result($con, $pageSize, true, $cookie);
+        }
+        $this->serverctrls = [
+            [
+                'oid' => \LDAP_CONTROL_PAGEDRESULTS,
+                'isCritical' => true,
+                'value' => [
+                    'size' => $pageSize,
+                    'cookie' => $cookie,
+                ],
+            ],
+        ];
+
+        return true;
+    }
+
+    /**
+     * Retrieve LDAP pagination cookie.
+     *
+     * @param resource $con
+     * @param resource $result
+     *
+     * @return string
+     */
+    private function controlPagedResultResponse($con, $result, string $cookie = '')
+    {
+        if (\PHP_VERSION_ID < 70300) {
+            ldap_control_paged_result_response($con, $result, $cookie);
+
+            return $cookie;
+        }
+        $controls = [];
+        $errcode = 0;
+        ldap_parse_result($con, $result, $errcode, null, null, [], $controls);
```

Comment:
vars are passed by reference, this means that:
- there is no need to init any of them
- putting an inline value (`null, null, []`) is illegal AFAIK

Reply:
Not sure what you mean by `there is no need to init any of them`.
But that's what I've done:

```php
$controls = $errcode = $matcheddn = $errmsg = $referrals = null;
ldap_parse_result($con, $result, $errcode, $matcheddn, $errmsg, $referrals, $controls);
```